### PR TITLE
🌷 Website Maintenance

### PR DESCRIPTION
**Pull request by jescalan**
A bit more maintenance:

- Dependency updates including `next@9.4.x`
- Linting updates, including an eslint and stylelint run across all files
- `jsconfig.json` file allows absolute imports
- `fetch` polyfill removed as it comes out of the box with nextjs now
- `exportTrailingSlash` removed, which should enable custom 404 page and simplify routing
- styles added for custom 404 page
- Updates to consent manager to add all analytics scripts in a way that is compliant with GDPR
- Improves the product download component to default to 64-bit architecture for `x86_64` arch tags, removes some repeated processing code on the downloads page
- Standardizes and documents components available for use within markdown files, adding a new `Tabs` component
- Documents syntax highlighting standards
- You can now `@include` non-markdown files, they will be imported into a code block. This is great for things like code examples!

### CODE REVIEW

File: website/README.md [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
 This subdirectory contains the entire source for the [Nomad Website](https://nomadproject.io/). This is a [NextJS](https://nextjs.org/) project, which builds a static site from these source files.
 
+<!--
+  This readme file contains several blocks of generated text, to make it easier to share common information
+  across documentation website readmes. To generate these blocks from their source, run `npm run generate:readme`
+
+  Any edits to the readme are welcome outside the clearly noted boundaries of the blocks. Alternately, a
+  block itself can be safely "forked" from the central implementation simply by removing the "BEGIN" and
+  "END" comments from around it.
+-->
```
Comment (DingoEatingFuzz) [inline website/README.md:14]: Woah, slick tech.

File: website/pages/partials/envvars.mdx [hunk 2]
```diff
@@ -31,7 +31,7 @@
         <code>NOMAD_SECRETS_DIR</code>
       </td>
       <td>
-        Path to the task's secrets directory. See
+        Path to the task&lsquo;s secrets directory. See
```
Comment (DingoEatingFuzz) [inline website/pages/partials/envvars.mdx:34]: Are these `&lsquo;` part of a linter rule? Are they automatically applied?

Comment (jescalan) [reply website/pages/partials/envvars.mdx:34]: Yes they are, and they are not automatically applied. I am a little annoyed by them and a few others on the team are as well, but it was a recent addition to the eslint react defaults, so I'm trusting that it's there for a good reason 😕

File: website/README.md (hunk 1 above)
Comment (jescalan) [reply website/README.md:14]: 🎉 [This is the origin](https://github.com/hashicorp/nextjs-scripts#markdown-blocks) - `npm run generate:readme` will regenerate blocks any time!

**Comment by pruett** [comment]
Working on getting these updates merged in to prevent blocking ongoing work

**Pull request merged**